### PR TITLE
Replaced _VF formats with copy flags.

### LINKED ISSUE #3552: Core dump with /nsc /gdi:sw
When testing with /gdi:sw /nsc, it Segfault at

#0  WriteColor (color=790580992, format=1610877064, dst=0x8000ee7cec30 <Address 0x8000ee7cec30 out of bounds>)
    at /home/zihao/workspace/zihao_FreeRDP/include/freerdp/codec/color.h:791
#1  freerdp_image_copy (pDstData=0x7fffee7d0030 ":\036b", DstFormat=1610877064, nDstStep=<optimized out>, nXDst=<optimized out>, nYDst=<optimized out>, nWidth=1280,
    nHeight=672, pSrcData=0x7fffe66fb010 ":\036b", SrcFormat=537168008, nSrcStep=<optimized out>, nXSrc=0, nYSrc=0, palette=0x0)
    at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/codec/color.c:607
#2  0x00007ffff66e3b64 in gdi_surface_bits (cmd=0x6cb8a8, context=<optimized out>) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/gdi/gdi.c:983
#3  gdi_surface_bits (context=<optimized out>, cmd=0x6cb8a8) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/gdi/gdi.c:946
#4  0x00007ffff6737685 in update_recv_surfcmd_surface_bits (length=<synthetic pointer>, s=0x7fffe808aa60, update=0x6cb3b0)
    at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/surface.c:65
#5  update_recv_surfcmds (update=0x6cb3b0, size=275736, s=0x7fffe808aa60) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/surface.c:108
#6  0x00007ffff67354fd in fastpath_recv_update (fastpath=<optimized out>, updateCode=4 '\004', size=<optimized out>, s=0x7fffe808aa60)
    at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/fastpath.c:289
#7  0x00007ffff67364f2 in fastpath_recv_update_data (s=0x6c67e0, fastpath=0x6d33c0) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/fastpath.c:527
#8  fastpath_recv_updates (fastpath=0x6d33c0, s=0x6c67e0) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/fastpath.c:563
#9  0x00007ffff673185d in rdp_recv_fastpath_pdu (s=0x6c67e0, rdp=0x6b8900) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/rdp.c:1265
#10 rdp_recv_pdu (rdp=0x6b8900, s=0x6c67e0) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/rdp.c:1273
#11 0x00007ffff6732063 in rdp_recv_callback (transport=<optimized out>, s=0x6c67e0, extra=0x6b8900) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/rdp.c:1419
#12 0x00007ffff6739555 in transport_check_fds (transport=0x6c6480) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/transport.c:871
#13 0x00007ffff6732948 in rdp_check_fds (rdp=0x6b8900) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/rdp.c:1478
#14 0x00007ffff671fd31 in freerdp_check_fds (instance=0x6b3620) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/freerdp.c:264
#15 0x00007ffff67209c1 in freerdp_check_event_handles (context=0x6b38b0) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/core/freerdp.c:312
#16 0x0000000000418cc8 in xf_client_thread (param=0x6b3620) at /home/zihao/workspace/zihao_FreeRDP/client/X11/xf_client.c:1538
#17 0x00007ffff5f20d86 in thread_launcher (arg=0x6f90d0) at /home/zihao/workspace/zihao_FreeRDP/winpr/libwinpr/thread/thread.c:320

(gdb) f 2
#2  0x00007ffff66e3b64 in gdi_surface_bits (cmd=0x6cb8a8, context=<optimized out>) at /home/zihao/workspace/zihao_FreeRDP/libfreerdp/gdi/gdi.c:983
983                             if (!nsc_process_message(context->codecs->nsc, cmd->bpp, cmd->width,
(gdb) p *cmd
$6 = {cmdType = 0, destLeft = 0, destTop = 16, destRight = 1280, destBottom = 688, bpp = 32, codecID = 1, width = 1280, height = 672, bitmapDataLength = 275708,
  bitmapData = 0x7fffe60b502e "\005\271\002", skipCompression = false}

### PR DESCRIPTION
**Pull request by akallabeth**
* Do not encode vertical flip in color format (screws up if copied twice)
* Extend ```freerdp_image_copy``` with copy flipped flag (currently only vertical source implemented)
* Adjust ```nsc_process_message``` to accept flip argument
* Fixes #3552

### CODE REVIEW

**Comment by realjiangms** [comment]
Pure code review looks good. So we now only support:
1. vFlip on src image
2. Src image with vFlip should always have nYSrc = 0

**Comment by realjiangms** [comment]
I just tested with /nsc, but the screen somehow still corrupted.

**Comment by realjiangms** [comment]
@akallabeth I forgot to restart my shadow server with new code. It works perfectly

File: client/X11/xf_gdi.c [hunk 1]
```diff
@@ -1030,24 +1030,24 @@ static BOOL xf_gdi_surface_bits(rdpContext* context,
 			break;
 
 		case RDP_CODEC_ID_NSCODEC:
-			format = FREERDP_VFLIP_PIXEL_FORMAT(gdi->dstFormat);
+			format = gdi->dstFormat;
```
Comment (mfleisz) [inline client/X11/xf_gdi.c]: Isn't the format variable quite useless now?
I think we can just pass gdi->dstFormat (or PIXEL_FORMAT_BGRX32 in the next case) to the functions instead.

File: include/freerdp/codec/color.h [hunk 2]
```diff
@@ -69,87 +60,68 @@
  */
 
 /* 32bpp formats */
-#define PIXEL_FORMAT_A8R8G8B8_F(_flip)	FREERDP_PIXEL_FORMAT(_flip, 32, FREERDP_PIXEL_FORMAT_TYPE_ARGB, 8, 8, 8, 8)
+#define PIXEL_FORMAT_A8R8G8B8_F()	FREERDP_PIXEL_FORMAT(32, FREERDP_PIXEL_FORMAT_TYPE_ARGB, 8, 8, 8, 8)
```
Comment (mfleisz) [inline include/freerdp/codec/color.h]: I think we can get rid of all the _F macros and just use FREERDP_PIXEL_FORMAT directly for the #defines.

File: libfreerdp/gdi/gdi.c [hunk 3]
```diff
@@ -978,24 +977,24 @@ static BOOL gdi_surface_bits(rdpContext* context,
 			break;
 
 		case RDP_CODEC_ID_NSCODEC:
-			format = FREERDP_VFLIP_PIXEL_FORMAT(gdi->dstFormat);
+			format = gdi->dstFormat;
```
Comment (mfleisz) [inline libfreerdp/gdi/gdi.c]: Again - I don't think format is needed here anymore

File: client/X11/xf_gdi.c (hunk 1 above)
Comment (akallabeth) [inline client/X11/xf_gdi.c]: @mfleisz well, preference. looks better this way to me.

File: libfreerdp/gdi/gdi.c (hunk 3 above)
Comment (akallabeth) [inline libfreerdp/gdi/gdi.c]: @mfleisz again, preference...

**Pull request merged**